azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -59,14 +62,21 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
         directly.
 
         """
-        try:
-            uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
+        uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
+
+        for _ in range(1, NUM_GS_FETCH_RETRIES + 1):
             self.xml_text = wire_client.fetch_config(uri, wire_client.get_header())
             xml_doc = parse_doc(self.xml_text)
-
             self.incarnation = findtext(xml_doc, "Incarnation")
-            self.expected_state = findtext(xml_doc, "ExpectedState")
+
             role_instance = find(xml_doc, "RoleInstance")
+            if role_instance:
+                break
+        else:
```

Comment:
Shouldn't this else be indented one more level?

Reply:
No, it's the same level as the for loop: https://github.com/Azure/WALinuxAgent/commit/7c2141494203e71c7163787f11a61656be29da20#

Reply:
lol wow, I wonder who approved it :p

Could you fix it here please? This is wrong right? There's no if for this else?

Reply:
It's not wrong :) https://book.pythontips.com/en/latest/for_-_else.html#else-clause
It's less common, but basically it can be used to distinguish between the for loop completing normally or completing due to a break.

Reply:
woah that's awesome! Had no idea!